### PR TITLE
It2 season least accurate team

### LINKED ISSUE #9: Season Stats - least_accurate_team
These methods each take a season id as an argument and return the values described below.

**Description**
Name of the Team with the worst ratio of shots to goals for the season

**Return Value**
String

### PR DESCRIPTION
**Pull request by hopesgit**
Thanks a ton, @guardiac

Methods added:

- Helper
1. shots_to_goals_ratio_per_team(takes season id)
1. total_goals_per_team_per_season(takes season id)
1. total_shots_per_team_per_season(takes season id)

- Season
1. least_accurate_team(season id)

closes #9

### CODE REVIEW

File: lib/stat_tracker.rb [hunk 1]
```diff
@@ -119,13 +118,18 @@ def games_won_per_team_for(season_id)
     end
   end
 
-  def shots_to_goals_ratio_per_team
-    total_shots_per_team.merge(total_goals_per_team){|team_id, shots, goals| (shots.to_f / goals).round(2)}
+  def shots_to_goals_ratio_per_team(season_id)
+    total_shots_per_team_per_season(season_id).merge(total_goals_per_team_for_season(season_id)){|team_id, shots, goals| (shots.to_f / goals).round(2)}
   end
```
Comment (AngelaGuardia) [inline lib/stat_tracker.rb]: I like how we've been ending these methods with "for_season(season_id)" maybe you could do that here.

File: lib/stat_tracker.rb [hunk 2]
```diff
@@ -119,13 +118,18 @@ def games_won_per_team_for(season_id)
...
+      result
+    end
+  end
+  def total_shots_per_team_per_season(season_id)
     @game_teams.reduce(Hash.new(0)) do |result, game_team|
-      result[game_team.team_id] += 1 if game_team.season == season_id
+      result[game_team.team_id] += game_team.shots if game_team.season == season_id
       result
     end
   end
```
Comment (AngelaGuardia) [inline lib/stat_tracker.rb]: same here! for_season instead of per season

**Review by AngelaGuardia — CHANGES_REQUESTED** [review]

File: lib/stat_tracker.rb [hunk 3]
```diff
@@ -236,6 +240,10 @@ def fewest_tackles(season_id)
     end[0]
   end
 
+  def least_accurate_team(season_id)
+    shots_to_goals_ratio_per_team(season_id).min_by { |season, avg| avg}[0].name
+  end
+
```
Comment (leahriffell) [inline lib/stat_tracker.rb:254]: @hopesgit I'm getting an error for this method here: _NoMethodError: undefined method `name' for "6":String_

File: test/stat_tracker_test.rb [hunk 4]
```diff
@@ -198,7 +190,7 @@ def test_it_can_return_total_games_per_team_for_a_season
     assert_equal ({"3"=>5, "6"=>9, "5"=>4, "17"=>1, "16"=>1}), @@stat_tracker.total_games_per_team_for("20122013")
   end
```
Comment (leahriffell) [inline test/stat_tracker_test.rb:192]: @hopesgit I'm getting an error for this test method: _NoMethodError: undefined method `total_games_per_team_for'_

**Review by leahriffell — CHANGES_REQUESTED** [review]
Hey Hope! Just left you a few comments on errors I get when running this branch.

File: lib/stat_tracker.rb (hunk 3 above)
Comment (hopesgit) [reply lib/stat_tracker.rb:254]: That's weird, I had that fixed before. It got un-fixed.

Comment (leahriffell) [reply lib/stat_tracker.rb:254]: Thanks, Hope!!

**Review by leahriffell — APPROVED** [review]

**Review by sarynm12 — APPROVED** [review]

File: test/stat_tracker_test.rb (hunk 4 above)
Comment (hopesgit) [reply test/stat_tracker_test.rb:192]: Yeah, I think the method got overwritten in the process of changing a few old helper methods around, so I just pulled it from the current main and put it back in.

File: lib/stat_tracker.rb (hunk 2 above)
Comment (hopesgit) [reply lib/stat_tracker.rb]: All fixed!

File: lib/stat_tracker.rb (hunk 1 above)
Comment (hopesgit) [reply lib/stat_tracker.rb]: And done!

**Review by AngelaGuardia — APPROVED** [review]

**Pull request merged**